### PR TITLE
Update sklearnex estimators to support sklearn 1.5

### PR DESCRIPTION
**Pull request by Alexsandruss**
Changes related to sklearn 1.5:
 - Apply renaming of `n_iter` parameter to `max_iter` in TSNE
 - Apply array API support in PCA internal functions
 - Apply `covariance_eigh` solver support in PCA
 - Change PCA "auto" solver choice to complete compliance with sklearn

### CODE REVIEW

File: sklearnex/decomposition/pca.py [hunk 1]
```diff
@@ -35,6 +35,9 @@
     if sklearn_check_version("1.1") and not sklearn_check_version("1.2"):
         from sklearn.utils import check_scalar
 
+    if sklearn_check_version("1.4"):
+        from sklearn.utils._array_api import get_namespace
```
Comment (icfaust) [inline sklearnex/decomposition/pca.py:39]:
```suggestion
    from sklearnex.utils import get_namespace

```

File: sklearnex/tests/test_n_jobs_support.py [hunk 2]
```diff
@@ -73,6 +74,9 @@ def check_methods_decoration(estimator):
     # by default, [Nu]SVC.predict_proba is restricted by @available_if decorator
     if estimator_class in [SVC, NuSVC]:
         estimator_kwargs["probability"] = True
+    # by default, sklearn 1.5 PCA selects non-patched method
+    if estimator_class == PCA:
+        estimator_kwargs["svd_solver"] = "full"
```
Comment (icfaust) [inline sklearnex/tests/test_n_jobs_support.py:79]: Ideally test_n_jobs_support should utilize sklearnex.tests._utils.SPECIAL_INSTANCES to make sure that the loss of a default check is covered by all of the test suite (and should be added there). This will save us headaches in the future.  Have we talked with @md-shafiul-alam about what we would need to do to support the default method?

Comment (icfaust) [reply sklearnex/tests/test_n_jobs_support.py:79]: Btw, I don't think its necessary for this PR to do it, but something we should plan for.

Comment (Alexsandruss) [reply sklearnex/tests/test_n_jobs_support.py:79]: It should be separate discussion/PR.
`SPECIAL_INSTANCES` has a specific behaviour: it initializes estimators once and will reuse them in different places. While estimator API theoretically allows this without any troubles, usage of same estimator instance in different tests might cause unexpected behaviour in future.

Comment (icfaust) [reply sklearnex/tests/test_n_jobs_support.py:79]: Thats absolutely true Maybe I should try to insert clone() at various places in the other tests using it.

Comment (icfaust) [reply sklearnex/tests/test_n_jobs_support.py:79]: Please look at #1801, which should address that concern.

File: sklearnex/decomposition/pca.py [hunk 3]
```diff
@@ -156,32 +162,38 @@ def transform(self, X):
...
                 U = U[:, : self.n_components_]
 
                 if self.whiten:
-                    U *= sqrt(X.shape[0] - 1)
+                    U *= sqrt(X_fit.shape[0] - 1)
                 else:
                     U *= S[: self.n_components_]
 
                 return U
+            else:
```
Comment (md-shafiul-alam) [inline sklearnex/decomposition/pca.py:194]: The current default method of sklearnex PCA is "covariance_eigh" although we call it "full". We don't have to fallback to sklearn. The _is_solver_compatible method can be updated to return true if the method=="covariance_eigh".

File: sklearnex/decomposition/pca.py [hunk 4]
```diff
@@ -129,7 +129,7 @@ def _onedal_fit(self, X, queue=None):
             onedal_params = {
                 "n_components": self.n_components,
                 "is_deterministic": True,
-                "method": "cov",
+                "method": "cov" if self._fit_svd_solver == "covariance_eigh" else "svd",
```
Comment (md-shafiul-alam) [inline sklearnex/decomposition/pca.py:132]: "svd" is slower than "cov" at the moment,  better to keep "cov" as default for now until "svd" is optimized. a separate investigation will be done to tune svd_solver for different data size

Comment (Alexsandruss) [reply sklearnex/decomposition/pca.py:132]: When svd and covariance solvers are simultaneously available in sklearn API, always defaulting to covariance is direct inconsistency with sklearn and disinforms users about used implementation.

**Comment by aepanchi** [comment]
I'm good with the doc changes

**Review by icfaust — APPROVED** [review]
This approval is contingent on the PCA discussion being completed (i.e. approval by @md-shafiul-alam or someone else with deeper knowledge on the PCA implementation). I think the other changes look good or have been discussed elsewhere (like the ensemble deselections).

File: sklearnex/decomposition/pca.py (hunk 4 above)
Comment (md-shafiul-alam) [reply sklearnex/decomposition/pca.py:141]: Agree on the inconsistency but the svd has been implemented very recently. Would be nice to have a perf measurement and decide on the trade-off.

Comment (md-shafiul-alam) [reply sklearnex/decomposition/pca.py:141]: BTW, "svd" is not an option for user input.

Comment (Alexsandruss) [reply sklearnex/decomposition/pca.py]: Considering oneDAL's SVD perf., I decided to choose next approach for svd_solver dispatching:
- Leave previous behaviour for sklearn<1.5
- If `full` svd_solver is explicitly requested fallback to stock sklearn for versions>=1.5
- If `auto` svd_solver selection of stock sklearn dispatches to `full` or `covariance_eigh`, use oneDAL PCA-Covariance with warning for `full` solver

Comment (md-shafiul-alam) [reply sklearnex/decomposition/pca.py]: The idea of explicit "full" dispatched to use SVD sounds good to me. Only thing is, oneDAL SVD solver should still outperform sklearn_svd_solver, unless you noticed otherwise. In that case, when explicit "full" is called, we can dispatch to onedal_svd.

Comment (Alexsandruss) [reply sklearnex/decomposition/pca.py]: oneDAL SVD performance improvement is not consistent on all cases/configuration, so it is better to leave it out of patch until SVD perf. is fixed/improved.

**Review by md-shafiul-alam — APPROVED** [review]
Looks good to me, the CI failures are addressed in https://github.com/intel/scikit-learn-intelex/pull/1812
